src/iologindata.cpp

```diff
@@ -160,6 +160,19 @@ uint32_t IOLoginData::gameworldAuthentication(const std::string& accountName, co
 	return accountId;
 }
 
+uint32_t IOLoginData::getAccountIdByPlayerName(const std::string& playerName)
```

Comment:
`bool House::isInvited(const Player* player)` is checked frequently, maybe needs a cache on Game class to improve performance.

Reply:
> `bool House::isInvited(const Player* player)` is checked frequently, maybe needs a cache on Game class to improve performance.

I like the idea, I'll try it thanks for the review

Reply:
i thought add ```uint32_t ownerAccount = 0;``` in House class, i tested and looks pretty good:

```c
// AccessHouseLevel_t House::getHouseAccessLevel(const Player* player)
	if (ownerAccount == player->getAccount()) {
		return HOUSE_OWNER;
	}
```


```diff
@@ void House::setOwner(uint32_t guid, bool updateDatabase/* = true*/, Player* player/* = nullptr*/)
{
        ....

		//clean access lists
		owner = 0;
+		ownerAccount = 0;
		setAccessList(SUBOWNER_LIST, "");
		setAccessList(GUEST_LIST, "");

    ....

	if (guid != 0) {
		std::string name = IOLoginData::getNameByGuid(guid);
		if (!name.empty()) {
			owner = guid;
			ownerName = name;
+			ownerAccount = IOLoginData::getAccountIdByPlayerName(name);
		}
	}
```

Reply:
It definitely looks great much better.